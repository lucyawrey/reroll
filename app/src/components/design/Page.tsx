import React, { ReactNode } from "react";
import { Container } from "react-bootstrap";

interface PageProps {
  children: ReactNode;
}

/**
 * The page component is used to wrap the content of a page to apply uniform
 * styling such as page margins.
 *
 * @param {ReactNode} props.children The child elements nested within this element.
 * These are passed automatically
 */
function Page(props: PageProps) {
  return (
    <Container className="mt-3">
      {props.children}
    </Container>
  );
}

export default Page;
