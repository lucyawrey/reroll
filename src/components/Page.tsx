import { makeStyles } from "@material-ui/styles";
import React, { ReactNode } from "react";

interface PageProps {
  children: ReactNode;
}

const useStyles = makeStyles({
  mainMargins: {
    margin: 15,
  },
});

/**
 * The page component is used to wrap the content of a page to apply uniform
 * styling such as page margins.
 *
 * @param {ReactNode} children The child elements nested within this element.
 * These are passed automatically
 */
function Page(props: PageProps) {
  const classes = useStyles();
  return (
    <main className={classes.mainMargins}>
      {props.children}
    </main>
  );
}

export default Page;
