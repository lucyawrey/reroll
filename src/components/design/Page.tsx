import React, { ReactNode } from "react";
import { Container } from "react-bootstrap";

interface PageProps {
  children: ReactNode;
  error?: object; // An error to render
}

function parseError(error: any): string {
  if (typeof error === "string") { return error; }
  try {
    return error.requestResult.responseContent.errors[0].cause[0].description;
  } catch {
    return "An unknown error has occured";
  }
}

/**
 * The page component is used to wrap the content of a page to apply uniform
 * styling such as page margins.
 *
 * @param {ReactNode} props.children The child elements nested within this element.
 * These are passed automatically
 */
export function Page(props: PageProps): JSX.Element {
  return (
    <Container className="mt-3">
      { props.error ? <>{parseError(props.error)}</> : <></>}
      {props.children}
    </Container>
  );
}

export default Page;
