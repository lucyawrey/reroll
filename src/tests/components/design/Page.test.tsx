import React from "react";
import renderer from "react-test-renderer";
import Page from "../../../components/design/Page";

describe('renders Page', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Page>Testing</Page>).toJSON()
    expect(tree).toMatchSnapshot()
  });
});
