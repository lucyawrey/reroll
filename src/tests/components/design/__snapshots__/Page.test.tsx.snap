// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Page renders correctly 1`] = `
<div
  className="mt-3 container"
>
  Testing
</div>
`;
